CommunicationLayer/MessagingFramework/MessageDecodingHelpers.cpp

```diff
@@ -27,23 +27,47 @@
 
 namespace
 {
-   const int NUMBER_OF_BYTES_IN_FLOAT = 4;
+    const int UNION_SIZE = 4;
 
-   union FloatCharTranslator
-   {
-      float floatData;
-      char charData[NUMBER_OF_BYTES_IN_FLOAT];
-   };
+    union DataFromCharTranslator
+    {
+        unsigned short unsignedShortData;
+        float floatData;
+        char charData[MessageDecodingHelpers::numberOfBytesInData(UNION_SIZE)];
+    }
 }
 
-float MessageDecodingHelpers::getFloat(
-   const QByteArray& data, int startIndex)
+float MessageDecodingHelpers::getFloat(const QByteArray& data, int startIndex)
 {
-   FloatCharTranslator dataUnion;
-   for (int i = 0; i < NUMBER_OF_BYTES_IN_FLOAT; i++)
-   {
-      dataUnion.charData[i] = data.at(i + startIndex);
-   }
+   return getData(data, startIndex, Type::FLOAT).floatData;
+}
+
+unsigned short MessageDecodingHelpers::getUnsignedShort(const QByteArray& data, int startIndex)
+{
+    return getData(data, startIndex, TYPE::UNSIGNED_SHORT).unsignedShortData;
+}
 
-   return dataUnion.floatData;
+void MessageDecodingHelpers::getData(const QByteArray& data, int startIndex, Type type)
+{
+    DataFromCharTranslator dataUnion;
+    for (int i = 0; i < numberOfBytesInData(type); i++)
+    {
+        dataUnion.charData[i] = data.at(i = startIndex);
+    }
+    return dataUnion;
+}
```

Comment:
@enochtsang I feel this could still be improved, but I wanted to get your take on it in case you would prefer a different direction? The big thing I don't like is 

`const int UNION_SIZE = 4;`

Because this could easily be missed by someone editing this file and cause problems down the road. I have the enum setup so I can iterate over it, should I be assigning this via logic and iterating over for the largest size, or do you have a more elegant solution?